### ISSUE TITLE
Kobo sync a book from the kobo store doesn't download the cover

### ISSUE THREAD

**Issue by JVT038** [janeczku/calibre-web#2371]
**Describe the bug/problem**
So I bought a book from the kobo store and I wanted to sync it.
I checked the 'Proxy unknown requests to Kobo Store' checkbox in the basic configurations, and my 'server external port' is set on 80.
The book is downloaded like it's supposed to, but it doesn't load / download the cover.

**Environment (please complete the following information):**
 - OS: Synology Diskstation Manager 7
 - Python version: 3.8.10
 - Calibre-Web version: 0.6.19 Beta - df53a5d8c97fe6790f75ed1d9e073b5f614dfc01 - 2022-04-13T18:37:23+02:00
 - Docker container: LinuxServer:
 - Special Hardware: Synology NAS DS918+
 - Browser: Firefox

**Additional context**
I access calibre-web through a reverse proxy

**Comment by jccpalmer** [comment]
Hey, did you ever get this resolved? I'm experiencing the same problem. I don't get covers from the store or Overdrive.

**Comment by JVT038** [comment]
Nope, no solution.

**Comment by flipside101** [comment]
Just wanted to add in im seeing this as well

**Comment by karlshea** [comment]
Looks like #2731 would fix this

**Comment by johnnyfleet** [comment]
I had the same issue and can now confirm that the above fix seems to have worked. Running [linuxserver/calibre-web:latest](https://hub.docker.com/r/linuxserver/calibre-web) docker instance and all works well now. 

Might be worth checking now @JVT038 .

**Comment by karlshea** [comment]
I don't think that docker image has the latest commits, when I look at contents the changes from 2731 aren't in there yet. I think there would need to be a new tagged release.

**Comment by ciceroii** [comment]
I'm running the most recent version, and still having the problem of the covers not showing. Is there anyway to debug why they might not showing?

**Comment by karlshea** [comment]
I have the new version and mine are also not working.

**Comment by rtrox** [comment]
It looks like Kobo changed their image endpoint, and this is currently broken again. From a fresh kobo sage's config:
```
image_host=https://cdn.kobo.com/book-images/
image_url_quality_template=https://cdn.kobo.com/book-images/{ImageId}/{Width}/{Height}/{Quality}/{IsGreyscale}/image.jpg
image_url_template=https://cdn.kobo.com/book-images/{ImageId}/{Width}/{Height}/false/image.jpg
```

**Comment by rtrox** [comment]
Opened a PR to resolve.

**Comment by johnnyfleet** [comment]
Thanks @rtrox. My earlier comment was wrong,  it didn't work then.

But I'm pleased that the latest nightly now works like a charm!

**Comment by jccpalmer** [comment]
The above solution worked for me, but then after syncing, I noticed it reverted back to my URL. Did anyone else notice this behavior or is it just me? If the latter, I must've done something wrong and if so, I wonder what.

**Comment by jccpalmer** [comment]
The above fix worked for me temporarily, but my config keeps reverting to my Calibre Web URL after a while. I can't figure out what causes it, as it seems to happen randomly after a sync or a reboot (but not always). I'm on the latest linuxserver Docker image that I just pulled yesterday (December 7). Luckily, I made a backup of the fixed config, so I can just delete the "broken" config file and get the fix when I need to. Annoying, but oh well.

Anyone else notice this or is it just me?

**Comment by rtrox** [comment]
@jccpalmer - when you say "the above fix", what do you mean? The only durable fix is to use the latest nightly of calibre-web. If you're just changing the `image_host` and other options in your kobo's config, calibre-web is going to overwrite those every time you sync, as it needs to be able to proxy the cover images so it can send you covers for your calibre library.

My PR is merged though, so if you update to the latest nightly of calibre-web, it will be able to proxy to the correct CDN url.

**Comment by mariolpantunes** [comment]
Hi, even with the latest docker image (0.6.24) from linux servers I cannot sync the cover from books bought from the kobo store.
Already checked that the new CDN is being used.

**Comment by mariolpantunes** [comment]
I was able to solve it, it was a misconfiguration in the reverse proxy.
